### ISSUE TITLE
Incorrect behavior of Map.resample_axis()

### ISSUE THREAD

**Issue by registerrier** [gammapy/gammapy#3102]
**Gammapy version**

dev

**Bug description**
`Map.resample_axis` does not return the same content if one removes one bin from the new axis.
For instance the outcome of `Map.resample_axis(axis)` will be different if one uses
`axis = MapAxis.from_edges((0.1,1.), name='energy', interp='log', unit='TeV')` or `axis = MapAxis.from_edges((0.1,1., 10.), name='energy', interp='log', unit='TeV')`.

**Expected behavior**
The content in the first bin should be identical.

**To Reproduce**
```
from gammapy.maps import MapAxis, WcsNDMap

axis = MapAxis.from_edges((0.1,1.0, 10.), name='energy', interp='log', unit='TeV')
axis2 = MapAxis.from_edges((0.1,1.), name='energy', interp='log', unit='TeV')
axis3 = MapAxis.from_edges((0.1,1., 10.), name='energy', interp='log', unit='TeV')

maps = WcsNDMap.create(npix=10, axes=[axis])
maps.data += 1
```

Then `maps.resample_axis(axis2).data` will yield:
```
array([[[2.]]], dtype=float32)
```
While `maps.resample_axis(axis2).data` will yield:
```
array([[[1.]],
       [[1.]]], dtype=float32)
```
**Other information**

**Comment by adonath** [comment]
Thanks @registerrier, I think this is because of the following behaviour of `np.add.reduceat`:
```python3
import numpy as np

np.add.reduceat([1, 2, 3, 4], indices=[0, 1])
```
Which returns:
```python3
array([1, 9])
```
So the last bin of the output array always sums up all contributions up to the last bin from the input array (see https://numpy.org/doc/stable/reference/generated/numpy.ufunc.reduceat.html).

I'm not sure what the best fix for this is...

**Comment by adonath** [comment]
Related: https://stackoverflow.com/questions/52319504/how-to-specify-the-last-index-explicitly-to-np-ufunc-reduceat

**Comment by adonath** [comment]
I think I can implement a fix for this...

**Comment by adonath** [comment]
Fixed by #3104.
